src/types.ts

```diff
@@ -23,3 +28,63 @@ export type Block = Record<string, BlockData>;
 export type BlockCache = Record<string, Block>;
 
 export type Cache = Record<number, BlockCache>;
+
+/**
+ * The interface for a service class responsible for making a request to an RPC
+ * endpoint.
+ */
+export type AbstractRpcService = {
```

Comment:
We could get this from `@metamask/network-controller`, but that would create a circular dependency.